src/client/js/CoreEditor.js

```diff
@@ -81,41 +82,57 @@ class CoreEditor extends React.Component {
         return editorState;
     }
     promptForLink() {
+
+        var title = this.getSelectedText();
+        
+        this.setState({
+            showURLInput: 1,
+            titleValue: title
+        });
+            
+    }
+    getSelectedText() {
         const {
             editorState
         } = this.state;
-        const selection = editorState.getSelection();
-        if (!selection.isCollapsed()) {
-            this.setState({
-                showURLInput: 1,
-                urlValue: ''
+        var title = '';
+        const selectionState = editorState.getSelection();
+        const blockKey = selectionState.getAnchorKey();
+        const contentBlocks = editorState.getCurrentContent().getBlocksAsArray();
+        var start = selectionState.getStartOffset();
+        var end = selectionState.getEndOffset();
+
+         contentBlocks.forEach((block) => {
+                if(block.key === blockKey){
+                    title = block.text.slice(start, end);
+                }
             });
-            }
+         return title;
     }
-    promptForEdit(urlValue, linkEntityKey) {
+    promptForEdit(titleValue, linkEntityKey) {
         const {
             editorState
         } = this.state;
         this.setState({
             showURLInput: 1,
             isEditLink: 1,
-            urlValue,
+            titleValue,
```

Comment:
@muddaserahmed are you sure you want rename urValue to titleValue? I think you will need to work with three link parameters (url,title,description). 
Now, if i enter link url and click confirm, and then try to edit this link, url and description are not saved inside the link